crates/wasm-encoder/src/core/branch_hints.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{CustomSection, Encode, Section, SectionId};
+use std::borrow::Cow;
+
+/// Helper structure to encode the `metadata.code.branch_hint` custom section.
+///
+/// This section was defined in the branch-hinting proposal for WebAssembly:
+/// <https://github.com/WebAssembly/branch-hinting>.
+#[derive(Default, Debug)]
+pub struct BranchHints {
+    bytes: Vec<u8>,
+    num_hints: u32,
+}
```

Comment:
Can you add an example to this doc comment, like the rest of the `wasm-encoder` sections have?